### ISSUE TITLE
Half of frames not rendered with polyscope::screenshot?

### ISSUE THREAD

**Issue by bbrrck** [nmwsharp/polyscope#65]
Consider the following program:
```cpp
#include <iostream>
#include <polyscope/polyscope.h>
int main(int argc, char* argv[]) {
  size_t frame_id = 0;
  polyscope::init();
  polyscope::state::userCallback = [&frame_id]() {
    std::cout << frame_id++ << " ";
  };
  polyscope::show(10);
  return 0;
}
```

As expected, this outputs `0 1 2 3 4 5 6 7 8 9 10`. The problem arises when I add a call to `polyscope::screenshot` in `userCallback`, as follows:

```cpp
#include <iostream>
#include <polyscope/polyscope.h>
int main(int argc, char* argv[]) {
  size_t frame_id = 0;
  polyscope::init();
  polyscope::state::userCallback = [&frame_id]() {
    std::cout << frame_id++ << " ";
    polyscope::screenshot(false);
  };
  polyscope::show(10);
  return 0;
}
```

This program only outputs `0 1 2 3 4 5` and produces six screenshots (`screenshot_000000.tga–screenshot_000005.tga`).

Is this a bug?

Edit: `screenshot_000005`, not `screenshot_000006`

**Comment by nmwsharp** [comment]
Thanks for the report!

I haven't verified yet, but this sounds like a bug. The `screenshot()` function triggers a rerender of the scene (to ensure its up-to-date), and I'd bet that for some reason the rerender is erroneously incrementing the frame counter in `show()`.

I'll push a fix as soon as I have time.... we probably need to adjust the control flow around `show()`, `screenshot()` and context management.

As a workaround in the meantime, this behavior should be consistent, so you can double the time in your `show()` call to get the desired number of frames.

**Comment by nmwsharp** [comment]
Should be fixed in 118bfa8f2046546ae39cbd21109117dbf1501bf9
